packages/react-table/src/demos/Table.md

```diff
@@ -908,15 +908,18 @@ class ColumnManagementAction extends React.Component {
                 aria-labelledby="table-column-management-item1"
                 isChecked={this.state.check1}
                 name="check1"
+                id="check1"
                 onChange={this.handleChange}
               />
+              <label htmlFor="check1">
```

Comment:
I feel that's something that the component should handle instead of the consumer. When DataListCheck is used the DataListItemCells component should be wrapped by a label. This is an implementation detail of DataListCheck that the users should not be bothered with. 
Secondly this behavior exists also in the expandable rows for table. The toggle button has to be clicked directly, I would expect that clicking anywhere on the row would expand the row. (not sure if i's the same issue)

Reply:
Hey @KKoukiou! Thanks for taking a look, that was definitely my first thought as well. I actually brought that up to the other team leads to determine if this should be a component or example level approach, but the difficulty here is all of the interactive elements that could be in the datalist. We wouldn't want clicking anywhere within a row to select the row, in general, since you could have other interactive elements there and the different ways it could be implemented. So the idea was to give control to consumers if they want to add a label in a situation like this or if they don't want that functionality in a more complex example like[ this example](https://www.patternfly.org/v4/components/data-list#checkboxes-actions-and-additional-cells). It's definitely really tough for us to balance all of the different potential use cases/allowing for flexibility while not over burdening the consumer. I'm very open to continuing this conversation on it though! @mcarrano, is there anything I missed in terms of why we were thinking of handling this at the demo/example level instead of at the component level?

Reply:
I can see you point here, I suggest to leave then how you have implemented it here, after all it's just an example, not an API that we are bound to, and if we figure out something else that works for the use case you describe above we can improve it in the future!